Source/Scene/GltfVertexBufferLoader.js

```diff
@@ -343,19 +339,33 @@ CreateVertexBufferJob.prototype.set = function (
 };
 
 CreateVertexBufferJob.prototype.execute = function () {
-  var result = this.typedArray;
-  if (this.dequantize) {
-    result = AttributeCompression.dequantize(
-      this.componentType,
-      this.type,
-      this.typedArray
+  this.vertexBuffer = createVertexBuffer(
+    this.typedArray,
+    this.dequantize,
+    this.componentType,
+    this.type,
+    this.count,
+    this.context
+  );
+};
+
+function createVertexBuffer(
+  typedArray,
+  dequantize,
+  componentType,
+  type,
+  count,
+  context
+) {
+  if (dequantize && componentType !== ComponentDatatype.FLOAT) {
```

Comment:
I added `componentType !== ComponentDatatype.FLOAT` to prevent floats from being dequantized. I think this is better than erroring out in `ResourceLoader`